src/timeline/CommunitiesModel.cpp

```diff
@@ -70,6 +92,17 @@ CommunitiesModel::data(const QModelIndex &index, int role) const
             return 0;
         case CommunitiesModel::Roles::Id:
             return "";
+        case CommunitiesModel::Roles::UnreadMessages: {
+            int total{0};
+            for (const auto &[id, info] : cache::getRoomInfo(cache::joinedRooms()))
```

Comment:
This is really, really slow. Every time I click on a room, this causes `900*900*2=1620000` database transactions and makes everything slow. There is a command to fetch all rooms at once, but I don't think you should even fetch all rooms here, we are already caching this stuff somewhere.